### ISSUE TITLE
wine: build wineStaging with 64bit support

### ISSUE THREAD

**Issue by bendlas** [NixOS/nixpkgs#7798]
Now that wine has been enabled for 64bit support, wineStaging is still 32bit only.
I have attempted to enable it for 64bit here: https://github.com/bendlas/nixpkgs/commit/87ca35b1ebcfd58aee019693d3de11f21ff3d22b

Unfortunately, (at least) the 32/64 bit build fails with a build error:
```
In file included from /tmp/nix-build-wineWow-1.7.42-staging.drv-0/wine-1.7.42/server/async.c:32:0:
/tmp/nix-build-wineWow-1.7.42-staging.drv-0/wine-1.7.42/server/file.h:147:1: error: unknown type name 'shmglobal_t'
 extern shmglobal_t *shmglobal;
 ^
In file included from /tmp/nix-build-wineWow-1.7.42-staging.drv-0/wine-1.7.42/server/request.h:26:0,
                 from /tmp/nix-build-wineWow-1.7.42-staging.drv-0/wine-1.7.42/server/async.c:33:
/tmp/nix-build-wineWow-1.7.42-staging.drv-0/wine-1.7.42/server/thread.h:91:5: error: unknown type name 'shmlocal_t'
     shmlocal_t            *shm;           /* thread local shared memory pointer */
     ^
Makefile:218: recipe for target 'async.o' failed
make[1]: *** [async.o] Error 1
make[1]: Leaving directory '/tmp/nix-build-wineWow-1.7.42-staging.drv-0/wine64/server'
```

@abbradar you packaged wine-staging, do you have a clue what's going on?

**Comment by abbradar** [comment]
Hm, no idea, I haven't encountered this before. I've looked quickly though PKGBUILD from AUR but there is nothing in particular that looks relevant.

**Comment by Stpnbs** [comment]
Hi there.
I had the same issue and did fix it running tools/make_requests which will recreate the needed file : 
include/wine/server_protocol.h
from
server/protocols.def

For more details, your wineStaging patchinstall.sh should run this automatically. But if you passed --no-autoconf or if the script was unable to locate tools/make_requests you have to do it manually.

My 2cts, hope this help.

**Comment by bendlas** [comment]
@Stpnbs thanks a lot for the pointer!
I just attempted to recreate the error, in order to try your fix. So I checked out the current unstable channel and cherry-picked the relevant wine commits (to save on all the rebuilds in current master). Curiously enough, it built just like that.

I will retry the build based on master, later on my home machine and see if I can reproduce the build error that way. In the meantime, I'd be very glad for anyone to reproduce my build failure and/or success:

This is the state, that just built for me: https://github.com/bendlas/nixpkgs/tree/unstable-wine64

This is the earlier failing build, rebased on current master, if the build failure was deterministic, hence introduced by a commit after unstable, it should still fail: https://github.com/bendlas/nixpkgs/tree/wine64
